modules/library/render/src/main/java/org/geotools/renderer/label/LabelPainter.java

```diff
@@ -672,19 +691,28 @@ public void paintCurvedLabel(LineStringCursor cursor) {
             }
 
             // draw halo and label
+            // extracting label first line component and compute is metrics
```

Comment:
I'm not sure about the line metrics changing per line. Maybe? But until we see an actual issue, I'd let it stay (the code was already there in @nmco version)